app/reaction.go

```diff
@@ -15,43 +15,46 @@ func (a *App) SaveReactionForPost(reaction *model.Reaction) (*model.Reaction, *m
 		return nil, err
 	}
 
-	if a.License() != nil && *a.Config().TeamSettings.ExperimentalTownSquareIsReadOnly {
-		var channel *model.Channel
-		if channel, err = a.GetChannel(post.ChannelId); err != nil {
+	channel, err := a.GetChannel(post.ChannelId)
+	if err != nil {
+		return nil, err
+	}
+
+	if channel.DeleteAt > 0 {
```

Comment:
This is the main change (and other exactly equal in other function), the rest of the code is transformed from normal error handling to idiomatic error handling.